test-app/src/main/kotlin/uk/gov/justice/digital/hmpps/testapp/service/SubjectAccessRequestService.kt

```diff
@@ -0,0 +1,24 @@
+package uk.gov.justice.digital.hmpps.testapp.service
+
+import org.springframework.stereotype.Service
+import uk.gov.justice.hmpps.kotlin.sar.HmppsPrisonSubjectAccessRequestService
+import uk.gov.justice.hmpps.kotlin.sar.HmppsSubjectAccessRequestContent
+import java.time.LocalDate
+
+@Service
+class SubjectAccessRequestService() : HmppsPrisonSubjectAccessRequestService {
+  override fun getPrisonContentFor(prn: String, fromDate: LocalDate?, toDate: LocalDate?) =
+    prn.takeIf { prn.startsWith("A") }?.let {
```

Comment:
Wouldn't it be better to throw some kind error if not implemented rather than return a 204 which would suggest the service has no data for the prisoner? 

It presumably should return a 209 if the service doesn't support PRN or CRN which is akin to not having implemented it

Reply:
or is this not automatically wired up? and if so is there a default implementation that would return a 209

Reply:
ah, saw the `@ConditionalOnBean(HmppsSubjectAccessRequestService::class)`, no worries.. so there would just be no endpoint.

Reply:
yup, nothing is wired up if the bean isn't defined